label_studio/projects/functions/next_task.py

```diff
@@ -176,17 +176,23 @@ def get_not_solved_tasks_qs(user, project, prepared_tasks, assigned_flag, queue_
             # don't output anything - just filter tasks with overlap
             logger.debug(f'User={user} tries overlap first from prepared tasks')
             _, not_solved_tasks = _try_tasks_with_overlap(not_solved_tasks)
-            queue_info += 'Show overlap first'
+            queue_info += (' & ' if queue_info else '') + 'Show overlap first'
 
-    return not_solved_tasks, user_solved_tasks_array, queue_info
+    return not_solved_tasks, user_solved_tasks_array, queue_info, prioritized_on_agreement
 
 
 def _prioritize_low_agreement_tasks(tasks, lse_project):
-    low_agreement_tasks = tasks.filter(_agreement__lt=lse_project.agreement_threshold, is_labeled=True)
-    return (True, low_agreement_tasks) if low_agreement_tasks else (False, tasks)
+    # if there are any tasks with agreement below the threshold which are labeled, prioritize them over the rest
+    # and return all tasks to be considered for sampling in order by least agreement
+    prioritized_low_agreement = tasks.filter(_agreement__lt=lse_project.agreement_threshold, is_labeled=True)
 
+    if prioritized_low_agreement.exists():
+        return True, tasks.order_by('-is_labeled', '_agreement')
 
-def get_next_task_without_dm_queue(user, project, not_solved_tasks, assigned_flag):
+    return False, tasks
+
+
+def get_next_task_without_dm_queue(user, project, not_solved_tasks, assigned_flag, prioritized_low_agreement):
```

Comment:
If you have the time, I think it'd be beneficial to put type hints on all the arguments and especially the return types of these functions (which btw are `Tuple[Thing1, Thing2, ..., ThingN]` it looks like

Reply:
I say this because even a local install of pyright / mypy would catch the bugs that could result from the return from a call to one of these functions being unpacked into too few variables.

Reply:
I can fast follow this, I would need a bit more time to understand how to do that (typing in python is something I haven't done). But 100% agree with the ask.